mod_form.php

```diff
@@ -121,10 +121,14 @@ public function definition() {
 
             foreach ($mycourses as $mycourse) {
                 $courselabel = $catlist[$mycourse->category] . ' / ' . $mycourse->fullname.' ('.$mycourse->shortname.')';
-                $options[$mycourse->id] = $courselabel;
                 if (empty($mycourse->visible)) {
-                    $hiddenlabel = ' '.get_string('hiddencourse', 'subcourse');
-                    $options[$mycourse->id] .= $hiddenlabel;
+                    if($config->displayhiddencourses || $mycourse->id == $currentrefcourseid){
+                        $hiddenlabel = ' '.get_string('hiddencourse', 'subcourse');
+                        $options[$mycourse->id] =  $courselabel.$hiddenlabel;
+                    }
+                }
```

Comment:
the else should be on the same line as the closing bracket:
```
} else {
```
Not:
```
}
else {
```

Reply:
ok i fixed lint. 
Do I need also to fix the outher parts of code that failing lint?

Reply:
all pull requests with improvements are welcome, but the main thing is to make sure a pull request doesn't add any new code that fails lint checks :-)

Reply:
Yeah but my code did not caused any NEW errors.
https://github.com/catalyst/moodle-mod_subcourse/actions/runs/6808426454

So you are saying that in order to contribute I need to fix the plugin to work without lint errors? FIXING YOUR CODE?

I am fixing errors from code that I did not made. 
And runinng again the tests. 
The only test failing appears to be a behat tests of completion. 

I removed the line to check if this is was a case of malformed test. Because I tested myself the case and it was working as it should.
https://github.com/EsdrasCaleb/moodle-mod_subcourse/actions/runs/9571702501/job/26389413751
And voala! it worked.

 So the problem that the condition in line 59 of completion_course.feature is malformed and causing the error (must have something to do with the theme change in 4.0). I could try to fix the behat test but I am not specialized in this..

Well here is a functionality that was requested one year ago, that my client is requesting also, if you could adapt my solution to your plugin I would be satisfied
https://github.com/catalyst/moodle-mod_subcourse/issues/42

Reply:
thanks @EsdrasCaleb - I've merged your main one in for the 403 branch - it does introduce one new lint error related to line endings, but I'll fix that manually myself.

you don't have to fix all the lint errors when working on moodle plugins, although if you do fix them it will make a developers job easier. :-)

the main thing to be aware of is that when contributing work upstream you need to try and do it on the "latest" branch available in the upstream repo first - then once that's approved and working the developer may backport it to older branches.

Reply:
This versions has no lint erros anymore . But that 403 has

Reply:
the github actions aren't triggering any codecheckout lint errors that I could see on the latest branch, although there are still the failing behat tests which have been faliing for a while. 
![Screenshot from 2024-06-19 13-37-20](https://github.com/catalyst/moodle-mod_subcourse/assets/362798/d4fe9d1a-3eab-4303-b14e-06d74f0c76b1)

Reply:
thanks for sending through the PR! :-)